src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,126 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final double ARRAY_GROWTH = 1.5;
+    private static final int START_SIZE = 0;
+    private int size;
+    private Object[] array;
+
+    public ArrayList() {
+        array = new Object[DEFAULT_CAPACITY];
+        size = START_SIZE;
+    }
+
     @Override
     public void add(T value) {
-
+        checkSize();
+        array[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        checkIndexInAdd(index);
+        checkSize();
+        Object[] newArray = new Object[array.length];
+        System.arraycopy(array, 0, newArray, 0, index);
+        newArray[index] = value;
+        System.arraycopy(array, index, newArray, index + 1, array.length - index - 1);
+        array = newArray;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        int newSize = size + list.size();
+        if (array.length < newSize) {
+            Object[] newArray = new Object[newSize];
+            System.arraycopy(array, 0, newArray, 0, array.length);
+            array = newArray;
+        }
```

Comment:
we have separate method `grow` to increase capacity of our array, you can pass some `minSize` to it 
or reuse your `add(T value)` method